lib/rails_4_session_flash_backport/rails2/flash_hash.rb

```diff
@@ -31,7 +31,14 @@ def self.from_session_value(value)
 
       def to_session_value
         return nil if empty?
-        discard = @used.select { |key, used| used }.keys
+        with_values = @used.select { |key, used| used }
+
+        if Gem::Version.new(RUBY_VERSION) <= Gem::Version.new('1.8.7') ||
+           Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.0.0')
```

Comment:
From the description, 1.9.0 would be broken with this change. Not that I expect many people would be using such a version. Perhaps we could do away with the four `Gem::Version` instances and check whether the `select`ed results `respond_to?(:to_h)`?

Will push up an alternative approach.